### PR TITLE
Improving unit test stability on Travis CI

### LINKED ISSUE #242: Moving AProjectTestCase to concept.unittest.utils
The idea is to move `AProjectTestCase` to `de.dlr.sc.virsat.concept.unittest.util` so that we don't need to import package `de.dlr.sc.virsat.project.test` in our test fragments.

### LINKED ISSUE #250: Disable Logging on Travis CI
Currently our tests get stuck quite often. The output is topped somewhere in the middle. This could be due to the 4MB log size restriction of Travis-CI:

https://docs.travis-ci.com/user/common-build-problems/

We try to disable logging, maybe it has some effect on stability as well.

### LINKED ISSUE #253: Don't react to checksum errors on build
Travis CI switched to goole mirror of maven cetral. In some cases this creates checksum issues. Our builds are failing.

We should filter for these issues such as:

```
15:[WARNING] Checksum validation failed, expected 7708e3725564364556fdec9305bb6165a7ab0710 but is 5f4f224c8761270824d67b1d5b976de2c75e921f from google-maven-central for https://maven-central.storage-download.googleapis.com/repos/central/data/org/codehaus/mojo/build-helper-maven-plugin/1.9/build-helper-maven-plugin-1.9.pom
16:[WARNING] Could not validate integrity of download from https://maven-central.storage-download.googleapis.com/repos/central/data/org/codehaus/mojo/build-helper-maven-plugin/1.9/build-helper-maven-plugin-1.9.pom: Checksum validation failed, expected 7708e3725564364556fdec9305bb6165a7ab0710 but is 5f4f224c8761270824d67b1d5b976de2c75e921f
17:[WARNING] Checksum validation failed, expected 7708e3725564364556fdec9305bb6165a7ab0710 but is 5f4f224c8761270824d67b1d5b976de2c75e921f from google-maven-central for https://maven-central.storage-download.googleapis.com/repos/central/data/org/codehaus/mojo/build-helper-maven-plugin/1.9/build-helper-maven-plugin-1.9.pom
20:[WARNING] Checksum validation failed, expected 5836bbd95869cd3c62e0f9b9d3dd9c1189b825f5 but is fe624c05a53b279784987afd73434f1f5d5e0a3b from google-maven-central for https://maven-central.storage-download.googleapis.com/repos/central/data/org/codehaus/mojo/mojo-parent/33/mojo-parent-33.pom
21:[WARNING] Could not validate integrity of download from https://maven-central.storage-download.googleapis.com/repos/central/data/org/codehaus/mojo/mojo-parent/33/mojo-parent-33.pom: Checksum validation failed, expected 5836bbd95869cd3c62e0f9b9d3dd9c1189b825f5 but is fe624c05a53b279784987afd73434f1f5d5e0a3b
22:[WARNING] Checksum validation failed, expected 5836bbd95869cd3c62e0f9b9d3dd9c1189b825f5 but is fe624c05a53b279784987afd73434f1f5d5e0a3b from google-maven-central for https://maven-central.storage-download.googleapis.com/repos/central/data/org/codehaus/mojo/mojo-parent/33/mojo-parent-33.pom
```

### PR DESCRIPTION
**Pull request by ghost**
Initially the plan was to:

- Reduced logging size, since logs are restricted to 4MB
- Running unit tests for eac individual unit test fragment rather than all togeter in overall tesuite. Still the overall testuite is needed for downstream projects and fo executing tests in eclipse.
- Fixed issues with tests, and test confusion. every concept or plugin should have a test fragment. AllTests is suite for tests. In particular the surefire settings are realy sensitive to these settings. 
- Fixed several dependency issues  which became visible with the individual test execution.

Closes #250

Further along the way:
- We had to deal with checksum issues (Closes #253)
- Worked on dependecies on AProjectTestCase. Replaced all usages with AConceptProjectTestCase. Moving as suggested by #242 is not possible. (Closes #242)
- Fixed a bug on VirSatEditingDomain regarding the notification thread. In some test cases the thread was stopped making the test cases get stuck. Could be a reason to general test case instability we have seen in the past.
- Fixed issues with mechanical.test execution. Due to the individual teste executions  Super User Rightsw were not set, and therefore the inheritance copier failed to work as expected. Changed AProjectTestCase to now have SuperUser rights always being set. It seems to be the general case for tests.

### CODE REVIEW

File: de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java [hunk 1]
```diff
@@ -32,9 +32,13 @@
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.rules.TestName;
+<<<<<<< HEAD
+import org.junit.rules.Timeout;
+=======
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java]: Weird merge remains.

File: de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java [hunk 2]
```diff
@@ -80,6 +84,11 @@ protected void finished(Description description) {
 	protected static final String PROJECTNAME = "SWTBotTestProject";
 	protected IProject project;
 	
+	protected static final int MAX_TEST_CASE_TIMEOUT_SECONDS = 90;
+	
+	@Rule
+    public Timeout globalTimeout = Timeout.seconds(MAX_TEST_CASE_TIMEOUT_SECONDS);
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java]: Weird whitespacing.

**Review by SaMuellerDLR — CHANGES_REQUESTED** [review]

File: de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java (hunk 1 above)
Comment (PhilMFischer) [reply de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java]: Fixed. Did not pay attention when merging.

File: de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java (hunk 2 above)
Comment (PhilMFischer) [reply de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java]: Indeting was mixed by ws and tab. Fixed it.

**Review by SaMuellerDLR — APPROVED** [review]
Approved!

**Review by franzTobiasDLR — APPROVED** [review]

**Pull request merged**